cpp/src/utilities/high_res_timer.hpp

```diff
@@ -85,6 +85,17 @@ class HighResTimer {
     timers.clear();
   }
 
+  double average_runtime(std::string const &label)
```

Comment:
This is a bit inconsistent with the naming convention used here (verbs). Perhaps it should be `get_average_runtime()`. I would also move it above the "display" methods for consistency as well.

Reply:
Done.